src/checkers/inference/model/ComparableConstraint.java

```diff
@@ -5,25 +5,84 @@
 import org.checkerframework.framework.type.QualifierHierarchy;
 import org.checkerframework.javacutil.BugInCF;
 
+import com.sun.source.tree.Tree.Kind;
+
 /**
  * Represents a constraint that two slots must be comparable.
  *
  */
 public class ComparableConstraint extends Constraint implements BinaryConstraint {
 
+    private final ComparableOperationKind operation;
```

Comment:
Instead of creating two comparable constraints class, created an operation called ANY to indicate when the comparable constraint is a reference type. When the operation of the comparable constraints doesn't affect way comparable constraints are created.